esmvalcore/preprocessor/_derive/hurs.py

```diff
@@ -0,0 +1,45 @@
+"""Derivation of variable `hurs`."""
+
+from iris import NameConstraint
+import iris
+import numpy as np
+import cf_units
+
+from ._baseclass import DerivedVariableBase
+
+# Constants
+GAS_CONSTANT_WV = 461.5 # JK-1kg-1
+ENTALPY_OF_VAPORIZATION = 2.501e6 # Jkg-1
+
+class DerivedVariable(DerivedVariableBase):
+    """Derivation of variable `hurs`."""
+
+    @staticmethod
+    def required(project):
+        """Declare the variables needed for derivation."""
+        required = [
+            {'short_name': 'tdps'},
+            {'short_name': 'tas'},]
+        return required
+
+    @staticmethod
+    def calculate(cubes):
+        """Compute relative humidity.
+
+        Relative humidity computed from dewpoint temperature and 
+        surface air temperature following Bohren and Albrecht 1998.
+        """
+        tdps_cube = cubes.extract_cube(NameConstraint(var_name='tdps'))
+        tas_cube = cubes.extract_cube(NameConstraint(var_name='tas'))
+
+        cubes_difference = iris.analysis.maths.subtract(tas_cube, tdps_cube) 
```

Comment:
```suggestion
        cubes_difference = tas_cube - tdps_cube
```
Would this also work? Did you read the section on [cube maths](https://scitools-iris.readthedocs.io/en/stable/userguide/cube_maths.html) in the iris documentation?

Reply:
Thanks, so when I was trying it with simple `tas_cube-tdps_cube` the issue was that I was running out of memory, while it wasn't a case with `subtract`. I will test it again with the time diagnostic and will let you know.

Reply:
That would be surprising because it should do the same thing, i.e. the minus operator on an Iris cube is implemented using the exact same function:
https://github.com/SciTools/iris/blob/c5e0353da11c263656c6c77e5e56eafefcede077/lib/iris/cube.py#L3968